text_to_image/README.md

```diff
@@ -0,0 +1,79 @@
+# MLPerf™ Inference Benchmarks for Text to Image
+
+This is the reference implementation for MLPerf Inference text to image
+
+## Supported Models
+
+| model | framework | accuracy | dataset | model link | model source | precision | notes |
+| ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- |
+| StableDiffusion | Torch | - | Coco2014 | - | [Hugging Face](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0) | fp16 | NCHW||
+
+## Dataset
+TODO
+
+### Calibration Dataset
+
+
+## Setup
+Set the following helper variables
+```bash
+export ROOT=$PWD/inference
+export SD_FOLDER=$PWD/inference/text_to_image
+export LOADGEN_FOLDER=$PWD/inference/loadgen
+```
+### Clone the repository
+**TEMPORARLY:**
+```bash
+git clone --recurse-submodules https://github.com/pgmpablo157321/inference.git --branch stable_diffusion_reference --depth 1
+```
+**KEEP FOR LATER:**
+```bash
+git clone --recurse-submodules https://github.com/mlcommmons/inference.git --depth 1
+```
+Finally copy the `mlperf.conf` file to the stable diffusion folder
+```bash
+cp $ROOT/mlperf.conf $SD_FOLDER
+```
+
+### Install requirements (only for running without using docker)
+Install requirements:
+```bash
+cd SD_FOLDER
+pip install -r requirements.txt
+```
+Install loadgen:
+```bash
+cd LOADGEN_FOLDER
+CFLAGS="-std=c++14" python setup.py install
+```
+
+### Download dataset
+```bash
+cd $SD_FOLDER/tools
+./download-coco-2014.sh -n <number_of_workers>
+```
+For debugging you can download only a part of all the images in the dataset
+```bash
+cd $SD_FOLDER/tools
+./download-coco-2014.sh -m <max_number_of_images>
+```
+
+### Run the benchmark
+#### Local run
+```bash
+python3 main.py --dataset "coco-1024" --dataset-path coco2014 --profile stable-diffusion-xl-pytorch [--model-path <TODO: provide model weights>] [--dtype <fp32, fp16 or bf16>] [--device <cuda or cpu>] [--time 600] [--scenario SingleStream]
```

Comment:
While trying the below command 
```
CMD: /home/arjun/cm/bin/python3 main.py  --scenario Offline --profile stable-diffusion-xl-pytorch  --dataset coco-1024 --dataset-path /home/arjun/CM/repos/local/cache/03fbdcf95b3d4104/install --dtype fp32 --device cpu   --mlperf_conf '/home/arjun/CM/repos/local/cache/d31d3986d6ba4c43/inference/mlperf.conf'
```
this error is coming
```
ERROR:main:thread: failed on contentid=['/home/arjun/CM/repos/local/cache/03fbdcf95b3d4104/install/validation/data/COCO_val2014_000000214494.jpg'], The size of tensor a (128) must match the size of tensor b (21) at non-singleton dimension 3
```

Reply:
Thank you @pgmpablo157321 for checking. With `--threads 1` it is working fine and on RTX 4090 I'm seeing a mean latency of 5s.